learn.md

```diff
@@ -0,0 +1,185 @@
+
+### OpenMetrics
+
+[TBD, help wanted!](#how-to-contribute)
+
+Documentation is available [here](https://openmetrics.io/)
+
+### OpenTelemetry
+
+[TBD, help wanted!](#how-to-contribute)
+
+Documentation is available [here](https://opentelemetry.io/docs/)
+
+### OpenTracing
+
+[TBD, help wanted!](#how-to-contribute)
+
+Documentation is available [here](https://opentracing.io/docs/)
+
+### Prometheus 
+
+Prometheus is a single binary capable to collect, store, query and alert on your applications' metrics.
+It is designed for real time alerting and monitoring, with strong reliability guarantees. It has its own
+time series database build-in, allowing persistence across restarts and integrations with numerous external systems
+including CNCF [Cortex](#cortex), [Thanos](#thanos) and more. 
+
+Documentation is available [here](https://prometheus.io/docs/introduction/overview/)
+Best practices:
+
+* [Zen of Prometheus][zen-of-prom]
+
+[zen-of-prom]: https://the-zen-of-prometheus.netlify.app/
+ 
+#### Instrumenting Your Application With Metrics
+
+##### Granularity
+
+As asked [here](https://github.com/cncf/sig-observability/issues/18#issue-651759365), how much of cardinality is too much?
+
+Overall it depends on how you or other users plan to use those metrics, particularly what questions you will be asking your monitoring system.
+In terms of metrics, questions should be mostly around aggregated values over time like max, min, sum or avg.
+
+The amount of cardinality essentially depends on how much you want to pay for running Prometheus vs how many dimensions you want to your aggregate on.
+Usually it means avoiding unique, frequently changing info like IDs, IPs, user names, or error strings, so generally values with unbounded uniqueness. 
+Those are usually related to events, so logs or traces are more suitable to track each of those. The rule of thumb is checking if you can specify all
+possible values for such a label in programmatic `constant`. If because of the dynamic you cannot, it most likely should not be a metric.
+
+In terms of errors, there is still big value in knowing the "type" of error. Was it transient? Was it user error? Server error? Timeout? 
+It's very common to have a label for error type to at least have graphs showing the high-level reason like HTTP status code.
+
+##### Label vs Metric
+
+As asked [here](https://github.com/cncf/sig-observability/issues/18#issue-651759365), for example should 
+`flux_daemon_synced_manifests_total` be a single metric with the `status` label? Or should it be split in two; `flux_daemon_synced_manifests_failures_total`
+ and `flux_daemon_synced_manifests_success_total`?
+ 
+The rule of thumb is to ask yourself a question: Would you need to group the aggregations across those labels? If yes, put it in the label, if no then the metric name is fine.
+Remember that a new label means a bigger index, so some overhead and cost again. 
```

Comment:
This cost is generally trivial, and this sort of advice could lead to users incorrectly putting label values in metric names. It's cardinality that's expensive, not the labels themselves.

Reply:
Agree, added label overhead is not significant, will change - or ideally we can point to some place that explains it on Prom docs! (: